From e9a9de59bc306f81a26c59170b86ba6788858907 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 1 Sep 2014 20:15:27 -0400
Subject: [powerpc] perf: Add kconfig option for hypervisor provided counters

Message-id: <1409602534-23951-9-git-send-email-gduarte@redhat.com>
Patchwork-id: 90121
O-Subject: [RHEL7.1 PATCH BZ 947159 08/15] powerpc/perf: Add kconfig option for hypervisor provided counters
Bugzilla: 947159
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=947159
Upstream Status: 30daeb6c8f54841274d1845b85c41f44165b3755

commit 30daeb6c8f54841274d1845b85c41f44165b3755
Author: Cody P Schafer <cody@linux.vnet.ibm.com>
Date:   Fri Mar 14 16:00:43 2014 +1100

    powerpc/perf: Add kconfig option for hypervisor provided counters

    The commit adds a Kconfig option which allows the hv_gpci and hv_24x7
    PMUs, added in the preceeding commits, to be built.

    Signed-off-by: Cody P Schafer <cody@linux.vnet.ibm.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/perf/Makefile b/arch/powerpc/perf/Makefile
index 510fae1..98a91ae 100644
--- a/arch/powerpc/perf/Makefile
+++ b/arch/powerpc/perf/Makefile
@@ -11,5 +11,7 @@ obj32-$(CONFIG_PPC_PERF_CTRS) += mpc7450-pmu.o
 obj-$(CONFIG_FSL_EMB_PERF_EVENT) += core-fsl-emb.o
 obj-$(CONFIG_FSL_EMB_PERF_EVENT_E500) += e500-pmu.o
 
+obj-$(CONFIG_HV_PERF_CTRS) += hv-24x7.o hv-gpci.o hv-common.o
+
 obj-$(CONFIG_PPC64)  += $(obj64-y)
 obj-$(CONFIG_PPC32)  += $(obj32-y)
diff --git a/arch/powerpc/platforms/pseries/Kconfig b/arch/powerpc/platforms/pseries/Kconfig
index 62b4f80..3763418 100644
--- a/arch/powerpc/platforms/pseries/Kconfig
+++ b/arch/powerpc/platforms/pseries/Kconfig
@@ -110,6 +110,18 @@ config CMM
    will be reused for other LPARs. The interface allows firmware to
    balance memory across many LPARs.
 
+config HV_PERF_CTRS
+       bool "Hypervisor supplied PMU events (24x7 & GPCI)"
+       default y
+       depends on PERF_EVENTS && PPC_PSERIES
+       help
+   Enable access to hypervisor supplied counters in perf. Currently,
+   this enables code that uses the hcall GetPerfCounterInfo and 24x7
+   interfaces to retrieve counters. GPCI exists on Power 6 and later
+   systems. 24x7 is available on Power 8 systems.
+
+          If unsure, select Y.
+
 config DTL
  bool "Dispatch Trace Log"
  depends on PPC_SPLPAR && DEBUG_FS
-- 
1.7.1